x-pack/plugins/security/server/deprecations/privilege_deprecations.test.ts

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { GetDeprecationsContext } from 'src/core/server';
+
+import { getPrivilegeDeprecationsServices } from '.';
+import {
+  coreMock,
+  elasticsearchServiceMock,
+  loggingSystemMock,
+} from '../../../../../src/core/server/mocks';
+import { featuresPluginMock } from '../../../features/server/mocks';
+import { licenseMock } from '../../common/licensing/index.mock';
+import { AuthorizationService } from '../authorization';
+
+const kibanaIndexName = '.a-kibana-index';
+const application = `kibana-${kibanaIndexName}`;
+
+describe('#getPrivilegeDeprecationsServices', () => {
+  describe('#getKibanaRolesByFeatureId', () => {
+    const mockClusterClient = elasticsearchServiceMock.createClusterClient();
+    const mockAsCurrentUser = elasticsearchServiceMock.createScopedClusterClient();
+
+    const mockGetSpacesService = jest
+      .fn()
+      .mockReturnValue({ getSpaceId: jest.fn(), namespaceToSpaceId: jest.fn() });
+    const mockFeaturesSetup = featuresPluginMock.createSetup();
+    const mockLicense = licenseMock.create();
+    const mockCoreSetup = coreMock.createSetup();
+
+    const authorizationService = new AuthorizationService();
+    const getClusterClient = () => Promise.resolve(mockClusterClient);
+    const authz = authorizationService.setup({
+      http: mockCoreSetup.http,
+      capabilities: mockCoreSetup.capabilities,
+      getClusterClient,
+      license: mockLicense,
+      loggers: loggingSystemMock.create(),
+      kibanaIndexName,
+      packageVersion: 'some-version',
+      buildNumber: 42,
+      features: mockFeaturesSetup,
+      getSpacesService: mockGetSpacesService,
+      getCurrentUser: jest.fn(),
+    });
+
+    const { getKibanaRolesByFeatureId } = getPrivilegeDeprecationsServices(authz, mockLicense);
```

Comment:
Now you can simplify all of this quite a bit:

```suggestion
import { elasticsearchServiceMock } from '../../../../../src/core/server/mocks';
import { licenseMock } from '../../common/licensing/index.mock';

const kibanaIndexName = '.a-kibana-index';
const application = `kibana-${kibanaIndexName}`;

describe('#getPrivilegeDeprecationsServices', () => {
  describe('#getKibanaRolesByFeatureId', () => {
    const mockAsCurrentUser = elasticsearchServiceMock.createScopedClusterClient();
    const mockLicense = licenseMock.create();
    const authz = { applicationName: application };
    const { getKibanaRolesByFeatureId } = getPrivilegeDeprecationsServices(authz, mockLicense);
```